### PR TITLE
Fix storybook knobs with timemachine test helper and config file

### LINKED ISSUE #3896: Storybook knobs aren't working
**Describe the bug**
inputProvider service toggling isnt working

**To Reproduce**

1. Go to https://bbc.github.io/simorgh/?path=/story/containers-header--default
2. Flick through other services
3. See no change

NB: After investiating, it appears storybook `addon-knobs` are entirely broken in simorgh. They show, but changing them have no affect

**Expected behavior**
It changes for each service.

**Screenshots**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by hotinglok**
Resolves #3896 

**Overall change:** 
Fixed storybook knobs as well as added `withTimeMachine` testhelper for future components which make use of `timemachine`.

**Code changes:**
- Added `time-machine.js` and `withTimeMachine.jsx` testhelper.
- Replaced `timemachine` in `/.storybook/config.js` with `startTimeMachine()`.
- Wrapped relevant components with `<WithTimeMachine>` from testhelper.

**EDIT: Can't seem to reply to comments for some reason, timestamp stories were removed by accident, should be back now.**

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

### CODE REVIEW

**Review by thekp — COMMENTED** [review]
Why are you deleting `ArticlesTimestamp.index.stories.jsx` ? 

edit: 

~nvm, I just read the thread on slack.
I think we shouldn't delete it and add `.addParameters({ chromatic: { disable: true } })` to it if we don't want it checked on chromatic.~
e.g: https://github.com/bbc/simorgh/blob/latest/src/app/containers/Brand/index.stories.jsx#L10

nvm me again, it looks like ArticlesTimestamp already had that ^ configured

**Comment by oluoluoxenfree** [comment]
@lokh01 how can we test this without the timestamp stories?

**Comment by hotinglok** [comment]
@thekp @oluoluoxenfree added an edit in the original comment since I can't seem to reply to either of your comments. Removed by accident, added back now.

File: src/app/lib/utilities/moment/index.stories.jsx [hunk 1]
```diff
@@ -67,7 +68,6 @@ const OCTOBER = 1571097600000;
 const NOVEMBER = 1573776000000;
 const DECEMBER = 1576368000000;
 
-/* eslint-disable prettier/prettier */
```
Comment (thekp) [inline src/app/lib/utilities/moment/index.stories.jsx:70]: repost: https://github.com/bbc/simorgh/pull/4178/files#r333233667

Comment (hotinglok) [reply src/app/lib/utilities/moment/index.stories.jsx:70]: Misread it first time round, fixed now.

**Review by HarveyPeachey — COMMENTED** [review]
LGTM but might need some documentation or comments about what time machine does.

**Review by HarveyPeachey — APPROVED** [review]
LGTM

**Review by thekp — APPROVED** [review]
Nice work 💯 

but looks like you have some linting issues

**Review by thekp — APPROVED** [review]

**Review by FK78 — APPROVED** [review]

File: src/app/containers/RadioPageMain/index.stories.jsx [hunk 2]
```diff
@@ -2,10 +2,10 @@ import React from 'react';
 import { storiesOf } from '@storybook/react';
 import { inputProvider } from '@bbc/psammead-storybook-helpers';
 import { withKnobs } from '@storybook/addon-knobs';
+import WithTimeMachine from '../../../testHelpers/withTimeMachine';
```
Comment (ryanmccombe) [inline src/app/containers/RadioPageMain/index.stories.jsx]: Does the alias work here?
```suggestion
import WithTimeMachine from '#testHelpers/withTimeMachine';
```

File: src/app/lib/utilities/moment/index.stories.jsx [hunk 3]
```diff
@@ -10,6 +10,7 @@ import {
   GEL_SPACING_DBL,
 } from '@bbc/gel-foundations/spacings';
 import { GEL_FF_REITH_SANS } from '@bbc/gel-foundations/typography';
+import WithTimeMachine from '../../../../testHelpers/withTimeMachine';
```
Comment (ryanmccombe) [inline src/app/lib/utilities/moment/index.stories.jsx]: Does the alias work here?
```suggestion
import WithTimeMachine from '#testHelpers/withTimeMachine';
```

File: src/testHelpers/withTimeMachine.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+// This affects the global Date object for the storybook application, to ensure consistency in chromaticQA testing.
+const WithTimeMachine = ({ children }) => {
+  // set time, but leave it ticking, so knobs work
+  startTimeMachine(false);
+  // setting tick to false, prevents the timemachine from ticking,
+  // so the timestamp displayed is frozen, and there are no side effects.
+  useEffect(() => {
+    resetTimeMachine();
+    startTimeMachine(true);
+    return () => {
```
Comment (ryanmccombe) [inline src/testHelpers/withTimeMachine.jsx]: Wrapper needed?
```suggestion
return resetTimeMachine;
```

**Review by ryanmccombe — APPROVED** [review]

File: src/app/lib/utilities/moment/index.stories.jsx (hunk 3 above)
Comment (FK78) [reply src/app/lib/utilities/moment/index.stories.jsx]: Yes.

File: src/app/containers/RadioPageMain/index.stories.jsx (hunk 2 above)
Comment (FK78) [reply src/app/containers/RadioPageMain/index.stories.jsx]: Yes.

File: src/testHelpers/withTimeMachine.jsx (hunk 4 above)
Comment (lokh01) [reply src/testHelpers/withTimeMachine.jsx]: Functionality is still there without it, what does the wrapper do/improve on in this case?

Comment (hotinglok) [reply src/testHelpers/withTimeMachine.jsx]: Functionality is there without it, what does the wrapper do/improve on in this case?

Comment (ryanmccombe) [reply src/testHelpers/withTimeMachine.jsx]: I'm suggesting removing the wrapper you have, not adding another one :)

before:
```
return () => {
  resetTimeMachine ();
}
```
after:
```
return resetTimeMachine;
```

Comment (hotinglok) [reply src/testHelpers/withTimeMachine.jsx]: I don't even have the excuse that it's friday :P, nice catch, my b

**Review by j-pendlebury — APPROVED** [review]
Looks good 👍

**Review by EinsteinNjoroge — APPROVED** [review]

File: src/app/lib/utilities/moment/index.stories.jsx [hunk 5]
```diff
@@ -89,100 +90,664 @@ const editorialWhitelist = [
 ];
 
 const methods = [
-  { what: '[default format]', method: locale => moment(A_DATE).locale(locale).format() },
```
Comment (thekp) [inline src/app/lib/utilities/moment/index.stories.jsx]: the formatting messed up again 😢

File: src/testHelpers/withTimeMachine.jsx (hunk 4 above)
Comment (ryanmccombe) [reply src/testHelpers/withTimeMachine.jsx]: 👍 my two-word comment wasn't great

https://www.youtube.com/watch?v=_K-L9uhsBLM

**Review by thekp — APPROVED** [review]
Thanks for the fix 💯

**Pull request merged**